databroker/core.py

```diff
@@ -484,7 +541,7 @@ def to_dask(self):
     def _close(self):
         self.bag = None
 
-    def canonical(self, *, fill):
+    def canonical(self, *, fill, chunk_size=1):
```

Comment:
I think our options are:
* Require the user to specify a `chunk_size` or explicitly `chunk_size='auto'`. This is how `dask.array` used to work.
* Use `chunk_size='auto'` by default. This is how `dask.array` works nowadays.

In the absence of user input, I think we're likely to get the best performance from deferring to the server rather than imposing a fixed chunk size.